### PR TITLE
Notify transport that a TApplicationException is written

### PR DESCRIPTION
**Pull request by alandau**

### CODE REVIEW

File: swift-service/src/main/java/com/facebook/swift/service/ThriftMethodProcessor.java [hunk 1]
```diff
@@ -220,9 +222,11 @@ public void onFailure(Throwable t)
                         } else {
                             contextChain.undeclaredUserException(t);
```
Comment (jesboat) [inline swift-service/src/main/java/com/facebook/swift/service/ThriftMethodProcessor.java]: Does this set headers already? (I don't know)

Comment (alandau) [inline swift-service/src/main/java/com/facebook/swift/service/ThriftMethodProcessor.java]: This calls a user callback that can set headers. We have some predefined callback class that does.

Comment (alandau) [inline swift-service/src/main/java/com/facebook/swift/service/ThriftMethodProcessor.java]: But it doesn't cover the case of nifty explicitly sending a TAE. In those places, we don't have access to these contextChain callbacks

Comment (jesboat) [inline swift-service/src/main/java/com/facebook/swift/service/ThriftMethodProcessor.java]: ah, ok

**Comment by jesboat** [comment]
Leaving this one to andrewcox or someone else to review since I am not that familiar with swift-service

**Pull request merged**